File: 0205-isomorphic-strings/solution.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        
        if len(s) != len(t):
            return False
        
        if len(s) == 0 and len(t) == 0:
            return True
        
        mapCharsS = dict()
        mapCharsT = dict()
        
        for index in range(len(s)):
            if (s[index] not in mapCharsS) and (t[index] not in mapCharsT):
                mapCharsS[s[index]] = t[index]
                mapCharsT[t[index]] = s[index]
            elif (s[index] in mapCharsS):
                if mapCharsS[s[index]] == t[index]:
                    continue
                else:
                    return False
            elif (t[index] in mapCharsT):
                if mapCharsT[t[index]] == s[index]:
                    continue
                else:
                    return False
        
        return True
                
        
        

File: 0205-isomorphic-strings/solution.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) 
    {
        if(s.size() != t.size()) return false;
        const int n = s.size();
        unordered_map<char, char> STmap;
        unordered_map<char, char> TSmap;

        for(int i = 0; i < n; i++)
        {
            if(STmap.find(s[i]) == STmap.end() && TSmap.find(t[i]) == TSmap.end())
            {
                STmap[s[i]] = t[i];
                TSmap[t[i]] = s[i];
            }
            else if(STmap[s[i]] != t[i] || TSmap[t[i]] != s[i]) return false;
        }
        return true;
    }
};
